### DISCUSSION TITLE
Bounds and extents discovery

### DISCUSSION THREAD

**Discussion by laultman** [StereoKit/StereoKit#453] [General]
I am using a model management concept rather than a model hierarchy. My use case is simple, I need to enable/disable models at the frame draw level. My service is maintaining its own hierarchy of models. I am using remote rendering to display the scenes in HoloLens 2 device. My service is composed of edge compute for remoting and Azure services to manage the models and data.
It seems that I need to be able to actively wrap a bounds box around a collection of models so that I may treat them as a unit for add, remove, movement, scale, etc.

If this is already envisioned in SK, I can't find it or simply don't understand how some things work in SK. As you might imagine the concept is a collection of models may form a single object (rooms collection => house) while trees in the yard are not part of the house, but the house and trees together form a "lot". I want to move the house on the lot. I want to put the trees in different places. AND I want to move the lot around as a unit.

I think this implies an interface (possibly a selection handle or something in the models) that let the user move up and down the hierarchy to act on the objects at that hierarchy level.

**Comment by laultman** [comment]
I have all this working in my use case above. Now my next problem is model FromFile(). I have models of arbitrary size (X,Y,Z) from the model maker. I need to set the model to an explicit size, not scale. How do I get the size vector of the model as it exists from the maker? With that information I can calculate a scale that fits my whole model.

**Comment by maluoi** [comment]
Heyas! Sorry for the late reply on this :)

To get the `Model` of unknown size into a specific size, I often start with getting some sort of normalization scale value! In the file picker demo I use this code, which brings the model to a size of 1m. You can then multiply _this_ value by whatever size you ultimately want!
```csharp
modelScale = 1 / model.Bounds.dimensions.Magnitude;
```

That method used the magnitude of the dimensions to determine the 'size' of your `Model`, but that may not necessarily be what you want, depending on your situation. Another alternative might be:
```csharp
Vec3  dims            = model.Bounds.dimensions;
float max             = Math.Max(dims.x, Math.Max(dims.y, dims.z));
float normalizedScale = 1 / max;
```
